utils/src/array.rs

```diff
@@ -14,6 +14,8 @@ use thiserror::Error;
 pub enum Error<E: StdError + Send + Sync + 'static> {
     #[error("invalid bytes")]
     InsufficientBytes,
+    #[error("invalid u64 length")]
+    InvalidU64Length,
```

Comment:
Given this `Error` is used by anyone that implements `Array`, I think we should put this `U64Array` error in a different place.